libs/component-library/src/lib/tabs/tabs.tsx

```diff
@@ -1,60 +1,113 @@
-import { useTheme } from '@shopify/restyle';
 import React from 'react';
 import { FxBox, FxReanimatedBox } from '../box/box';
-import { FxPressableOpacity } from '../pressable-opacity/pressableOpacity';
+import {
+  FxPressableOpacity,
+  FxPressableOpacityProps,
+} from '../pressable-opacity/pressableOpacity';
 import { FxText, FxTextProps } from '../text/text';
-import { FxTheme } from '../theme/theme';
 import {
   useAnimatedReaction,
   useAnimatedStyle,
-  useDerivedValue,
   useSharedValue,
   withTiming,
 } from 'react-native-reanimated';
+import { FxSpacer } from '../spacer/spacer';
+import { StyleSheet, View } from 'react-native';
+import { ColorProps } from '@shopify/restyle';
+import { FxTheme } from '../theme/theme';
+
+const ANIMATION_DURATION = 150;
+
+type ColorOverrideType = ColorProps<FxTheme> & {
+  color: keyof FxTheme['colors'];
+};
+
+const SELECTED_TEXT: ColorOverrideType = { color: 'greenBase' };
+const AUTO_PRESSED: ColorOverrideType = { color: 'border' };
+const FIXED_PRESSED: ColorOverrideType = {
+  color: 'backgroundSecondary',
+};
 
-const ANIMATION_DURATION = 300;
-const SELECTED_TEXT: FxTextProps = { color: 'greenBase' };
-const PRESSED_TEXT: FxTextProps = { color: 'backgroundSecondary' };
+const FIXED_TAB_STYLES: FxPressableOpacityProps = {
+  borderBottomWidth: 2,
+  borderBottomColor: 'content3',
+  paddingVertical: '16',
+  hitSlop: 0,
+};
+
+const AUTO_TAB_STYLES: FxPressableOpacityProps = {
+  paddingVertical: '8',
+  hitSlop: 16,
+};
+
+const FIXED_TEXT_STYLES: FxTextProps = {
+  variant: 'bodySmallRegular',
+};
+
+const AUTO_TEXT_STYLES: FxTextProps = {
+  variant: 'bodyMediumRegular',
+};
 
-type FixedTabProps = {
+const VARIANT_STYLES = {
+  fixed: {
+    defaults: FIXED_TAB_STYLES,
+    defaultsText: FIXED_TEXT_STYLES,
+    pressedText: FIXED_PRESSED,
+    pressedBackground: {
+      backgroundColor: 'backgroundPrimary',
+      borderBottomColor: 'backgroundSecondary',
+    } as FxPressableOpacityProps,
+  },
+  auto: {
+    defaults: AUTO_TAB_STYLES,
+    defaultsText: AUTO_TEXT_STYLES,
+    pressedText: AUTO_PRESSED,
+    pressedBackground: {} as FxPressableOpacityProps,
+  },
+};
+
+type TabProps = {
   text: string;
   selected: boolean;
   onPress: () => void;
+  variant: 'auto' | 'fixed';
 };
 
-const FixedTab = ({ text, selected, onPress }: FixedTabProps) => {
-  const theme = useTheme<FxTheme>();
+const Tab = ({ text, selected, onPress, variant }: TabProps) => {
   const [pressed, setPressed] = React.useState<boolean>(false);
```

Comment:
Don't need to specify boolean here as implicit typing will infer it since starting states is set to false